client/state/notices/middleware.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { truncate } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { successNotice, errorNotice } from 'state/notices/actions';
+import { getSitePost } from 'state/posts/selectors';
+import {
+	POST_DELETE_FAILURE,
+	POST_DELETE_SUCCESS,
+	POST_SAVE_SUCCESS
+} from 'state/action-types';
+
+export const OBSERVERS = {
+	[ POST_DELETE_FAILURE ]: ( action, dispatch, getState ) => {
```

Comment:
> javascript decorators are still stage-1

we can already accomplish this goal with some code transformation, something @aduth I think you and I discussed with someone else (or more people too). the cool thing about using these macros (however you want to call them, come they from Babel or sweet.js or anywhere else) is that by keeping them in-syntax, we can make a system that works whether or not the macro is active

```js
const handlePostDeleteSuccess = ( … ) => …
createActionHandler( POST_DELETE_SUCCESS, handlePostDeleteSuccess );
```

we can write `createActionHandler` first as a simple JS function, then when we get fed-up or experimental, we can write a macro to transform that at compile-time to whatever the decorator would do.

I still vote to keep these handlers all centralized and in one place, even with the decorators.